openedx/features/course_experience/tests/test_course_updates.py

```diff
@@ -0,0 +1,121 @@
+"""
+Tests for the course updates utility methods.
+"""
+
+from django.test.client import RequestFactory
+
+from openedx.core.djangoapps.user_api.course_tag.api import get_course_tag, set_course_tag
+from openedx.features.course_experience.course_updates import (
+    dismiss_current_update_for_user, get_current_update_for_user, get_ordered_updates,
+)
+from openedx.features.course_experience.tests import CourseUpdatesTestCase
+
+
+class TestCourseUpdatesUtils(CourseUpdatesTestCase):
+    """Tests for the course update utility methods."""
+
+    UPDATES_TAG = 'view-welcome-message'
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        cls.request = RequestFactory().get('/')
```

Comment:
Why the `.get('/')`?

Reply:
I think that's just how the request factory works -- you give it a path to fake a request for. Since the path doesn't matter here, I just used `/`

Reply:
I think you might be able to not even do the `.get` at all. IIRC, the RequestFactory makes the request. I would actually guess that you're setting cls.request to a Response since it would be the result of doing the `.get`.

Reply:
Looking at the code, you need to do the `.get()` (or `.option()` or `.post()` etc) and it returns a `WSGIRequest`.

Reply:
Wild. Well better to not change too much needlessly when in this dark world.